src/rust/engine/graph/src/node.rs

```diff
@@ -32,7 +32,7 @@ pub trait Node: Clone + Debug + Display + Eq + Hash + Send + 'static {
   ///
   /// If the node result is cacheable, return true.
```

Comment:
Nit: one issue with this method signature though, is that it's not clear to the implementor of the method what to return when there is no result present. If you had two methods, it would be harder to get that wrong.